### ISSUE TITLE
Editor creates degenerate surface

### ISSUE THREAD

**Issue by ljbrackney** [NREL/openstudio-geometry-editor#48]
Simple two story building results in degenerate surface which causes failure on OS import.

[floorplan.json.zip](https://github.com/NREL/openstudio-geometry-editor/files/1033600/floorplan.json.zip)

**Comment by macumber** [comment]
Space is defined by a face that has one point (e.g. single point) or two points (e.g. a line), does not define a valid polygon with area.

**Comment by devKate** [comment]
Sorry, I'm a bit confused about the issue. I imported that floorplan and it loaded into the geometry editor the way I'd expect, but I'm not sure what happens during OS imports. Are there certain types of faces I need to validate against? I have already prevented creating faces with no area.

**Comment by ljbrackney** [comment]
This fails on import. Dan will have to provide guidance on validation gaps.

On Jun 6, 2017, at 10:20 AM, Katie Noland <notifications@github.com<mailto:notifications@github.com>> wrote:


Sorry, I'm a bit confused about the issue. I imported that floorplan and it loaded into the geometry editor the way I'd expect, but I'm not sure what happens during OS imports. Are there certain types of faces I need to validate against? I have already prevented creating faces with no area.

-

**Comment by macumber** [comment]
@devKate part of the issue is that OS needs to trap for degenerate surfaces in the json, if you have prevented degenerate (e.g. 0 area) surfaces then that is also good, Larry might have used an old build of this tool that did not have that check in it.  If we continue seeing degenerate surfaces we will push it back to you

**Comment by ljbrackney** [comment]
I was using the first "release" build.  I'm currently pulling from Dev and can re-test as needed with that floorplan.json or a fresh one.

**Comment by macumber** [comment]
@devKate it seems like bad vertices are being written to the file, this doesn't prevent the editor from re-opening the file but it does mess up openstudio when dealing with it.  See "Space 1 - 2" in the file below.  It is associated with face 33 which seems to have way more edges than it seems like it should.  These edges include vertices that are referenced more than twice (e.g. vertices "52", "26"), each vertex should be referenced exactly twice (appears in two edges).  These two vertices are practically right on top of each other so they should really be the same point.  So something went wrong in either a splitting, line snapping, or some other operation.  Bottom line is we should not end up with vertices that are this close to each other, they should have been merged into one vertex.  

Vertex 26:
"id": "26",
"x": -23.182523015762783,
"y": -3.9715466950267597,

Vertex 52:
"id": "52",
"x": -23.18252,
"y": -3.97155,

Face 33:
"id": "33",
"edge_ids": ["31", "32", "133", "99", "91", "127", "137", "135", "167", "145", "174", "175", "168", "173", "176", "186", "187", "80"],
"edge_order": [1, 1, 1, 0, 1, 0, 0, 0, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1]


[floorplan.json.txt](https://github.com/NREL/openstudio-geometry-editor/files/1056825/floorplan.json.txt)

**Comment by macumber** [comment]
@ljbrackney a temporary fix is committed at https://github.com/NREL/OpenStudio/commit/7956d190b58d42d94cf257158146c0bfc9546a70, you can switch to the va3c_updates to see if it helps you for demo purposes

**Comment by ljbrackney** [comment]
Your most recent commit to the import_floorplan measure unblocked me.  I will continue testing with more complex geometries to see how things hang together.  Thanks!
